core/src/Cuda/Kokkos_Cuda_Half.hpp

```diff
@@ -212,6 +225,98 @@ class half_t {
   KOKKOS_FUNCTION
   half_t(unsigned long long rhs) : val(cast_to_half(rhs).val) {}
 
+  // Atomics
+  KOKKOS_FUNCTION
+  void atomic_add(const volatile half_t src) const volatile {
```

Comment:
In `atomic_add(const volatile half_t src)`, `src` is being passed by value, so all the `const volatile` does here is mark the local variable `src` as `const volatile`, which I don't think we actually need.


Also, marking parameters as `volatile` is deprecated as of C++20 (and gcc10 and clang11 will warn).

Reply:
Thank you for your feedback, @nliber. I will close this PR and address this in #4156 where `atomic_add` has been renamed to `impl_volatile_add`.